openpaas-james/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/openpaas/james/common/LinagoraKeystoreSetMethodContract.scala

```diff
@@ -481,22 +522,247 @@ trait LinagoraKeystoreSetMethodContract {
-         |				"created": {
+         |  "sessionState": "${SESSION_STATE.value}",
+         |  "methodResponses": [
+         |    ["Keystore/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |        "created": {
          |          "K87": {
          |            "id": "$PGP_KEY_ID"
-         |          },
-         |          "K88": {
-         |            "id": "$PGP_KEY_ID"
          |          }
-         |				}
-         |			}, "c1"]
-         |	]
+         |        }
+         |    }, "c1"],
+         |    ["Keystore/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "destroyed": [
+         |        "$PGP_KEY_ID"
+         |      ]
+         |    }, "c2"]
+         |  ]
+         |}""".stripMargin)
+
+    assertThat(server.getProbe(classOf[JmapGuiceKeystoreManagerProbe])
+      .retrieveKey(BOB, KeyId(PGP_KEY_ID)))
+      .isEmpty
+
+  }
+
+  @Test
+  def keystoreSetDestroyWithBackReferenceShouldSucceed(server: GuiceJamesServer): Unit = {
+    val request = s"""{
+                     |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:pgp"],
+                     |  "methodCalls": [
+                     |    ["Keystore/set", {
+                     |      "accountId": "$ACCOUNT_ID",
+                     |      "create": {
+                     |        "K87": {
+                     |          "key": "$PGP_KEY_ARMORED"
+                     |        }
+                     |      }
+                     |    }, "c1"],
+                     |    ["Keystore/set", {
+                     |      "accountId": "$ACCOUNT_ID",
+                     |      "destroy": [
+                     |        "#K87"
+                     |      ]
+                     |    }, "c2"]
+                     |  ]
+                     |}""".stripMargin
+
+    val response = `given`()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |  "sessionState": "${SESSION_STATE.value}",
+         |  "methodResponses": [
+         |    ["Keystore/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "created": {
+         |        "K87": {
+         |          "id": "$PGP_KEY_ID"
+         |        }
+         |      }
+         |    }, "c1"],
+         |    ["Keystore/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "destroyed": [
+         |        "$PGP_KEY_ID"
+         |      ]
+         |    }, "c2"]
+         |  ]
+         |}""".stripMargin)
+
+    assertThat(server.getProbe(classOf[JmapGuiceKeystoreManagerProbe])
+      .retrieveKey(BOB, KeyId(PGP_KEY_ID)))
+      .isEmpty
+  }
+
+  @Test
+  def keystoreSetDestroyWithUnknownKeyIdShouldSucceed(): Unit = {
```

Comment:
Hmmmm.... really? Back to the core specs: https://jmap.io/spec-core.html#set

```
If an id given cannot be found, the update or destroy MUST be rejected with a notFound set error.
```
And that believe that's what we do with mailbox and email

Reply:
Well, idempotence is a nice behaviour too. What we can is that the stuff that the user wanted to delete is no longer here.

Also, read-before-write is a classic anti-pattern in Cassandra.

I'm okay being slightly off-spec regarding this (as it is a custom extension).

Reply:
I believe any custom extension should still respect the core specs at the end... But I understand the anti-pattern in Cassandra this generates as well.

Ok for now